nova_vm/src/ecmascript/builtins/text_processing/string_objects/string_prototype.rs

```diff
@@ -718,16 +718,135 @@ impl StringPrototype {
         // 6. If functionalReplace is false, Set replaceValue to ? ToString(replaceValue).
         let replace_string = to_string(agent, replace_value)?;
         // Everything are strings: `"foo".replace("o", "a")` => use rust's replace
+        let result =
+            s.as_str(agent)
+                .replacen(search_string.as_str(agent), replace_string.as_str(agent), 1);
+        Ok(String::from_string(agent, result).into_value())
+    }
+
+    /// ### [22.1.3.20 String.prototype.replaceAll ( searchValue, replaceValue )](https://tc39.es/ecma262/multipage/text-processing.html#sec-string.prototype.replaceall)
+    fn replace_all(agent: &mut Agent, this_value: Value, args: ArgumentsList) -> JsResult<Value> {
+        // 1. Let O be ? RequireObjectCoercible(this value).
+        let o = require_object_coercible(agent, this_value)?;
+
+        let search_value = args.get(0);
+        let replace_value = args.get(1);
+
+        // 2. If searchValue is neither undefined nor null, then
+        if !search_value.is_null() && !search_value.is_undefined() {
+            // a. Let isRegExp be ? IsRegExp(searchValue).
+            let is_reg_exp = false;
```

Comment:
I think this goes beyong my knowledge.
Maybe [I can try to implement this](https://tc39.es/ecma262/multipage/abstract-operations.html#sec-isregexp) a different PR but I would need some guidance.

I did in this way because doing a quick search or `todo!()` macro, you can find all remaining things to implement. So it's not lost.

Reply:
Yeah, it's not completely lost so this is fine :)